scripts/prepare/bundle.ts

```diff
@@ -69,7 +69,16 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
           util: path.resolve('../node_modules/util/util.js'),
         }),
       ],
-      external: [name, ...Object.keys(dependencies || {}), ...Object.keys(peerDependencies || {})],
+      external: [
+        name,
+        ...Object.keys(dependencies || {}),
+        ...Object.keys(peerDependencies || {}),
+        // it's a transitive dependency of `@storybook/theming` that we want to keep external
+        // because it uses `React.insertionEffect` conditionally in a way that doesn't conflict with bundlers
+        // the bundled version gets changed though and creates problems when a bundle gets consumed later
+        // while this is only a dep of `@storybook/theming`, it should be safe to just add this here, for all packages
+        '@emotion/use-insertion-effect-with-fallbacks',
```

Comment:
Won't work for `lib/ui` because that needs to be an end-product (run directly in the browser.

I'd rather not have package specific exceptions in this bundle script.

Reply:
Ok, I've moved this to `@storybook/theming` dependencies